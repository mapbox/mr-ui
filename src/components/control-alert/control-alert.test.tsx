import React from 'react';
import ControlAlert from './control-alert';
import { screen, render, fireEvent } from '@testing-library/react';

describe('ControlAlert', () => {
  describe('basic error', () => {
    const props = {
      children: <span className="ml6">Drats.</span>,
      theme: 'error'
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlAlert {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('basic locked', () => {
    const props = {
      children: <span className="ml6">Nope.</span>,
      theme: 'locked'
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlAlert {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('basic success', () => {
    const props = {
      children: <span className="ml6">Woohoo!</span>,
      theme: 'success'
    } as const;

    test('renders as expected', () => {
      const { baseElement } = render(<ControlAlert {...props} />)
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('dismissive warning', () => {
    const mockOnClick = jest.fn();

    const props = {
      children: <span className="mx6">Uh oh...</span>,
      onButtonClick: mockOnClick,
      theme: 'warning'
    } as const;

    test('onButtonClick is called', () => {
      render(<ControlAlert {...props} />)
      fireEvent.click(screen.getByTestId('alert-dismiss'));
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('dismissive multiline error', () => {
    const mockOnClick = jest.fn();

    const props = {
      children: (
        <p className="mx6">
          This is the song that never ends.{' '}
          <a className="link" href="http://www.mapbox.com">
            Yes, it just goes on and on my friends.
          </a>{' '}
          Some people started singing it, not knowing what it was. And they'll
          continue singing it forever just because. This is the song that never
          ends.
        </p>
      ),
      onButtonClick: mockOnClick,
      theme: 'error'
    } as const;

    test('onButtonClick is called', () => {
      render(<ControlAlert {...props} />)
      fireEvent.click(screen.getByTestId('alert-dismiss'));
      expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
  });
});
