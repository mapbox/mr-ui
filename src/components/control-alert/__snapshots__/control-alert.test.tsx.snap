// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlAlert basic error renders as expected 1`] = `
<body>
  <div>
    <div
      class="round px24 py24 flex bg-red-faint"
    >
      <div
        class="flex-child-no-shrink"
      >
        <svg
          aria-hidden="true"
          class="events-none icon inline-block align-t"
          data-testid="icon-alert"
          focusable="false"
          style="width: 1.3em;"
        >
          <use
            xlink:href="#icon-alert"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
        >
          alert
        </span>
      </div>
      <div
        class="flex-child-grow"
      >
        <span
          class="ml6"
        >
          Drats.
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlAlert basic locked renders as expected 1`] = `
<body>
  <div>
    <div
      class="round px24 py24 flex bg-yellow-faint"
    >
      <div
        class="flex-child-no-shrink"
      >
        <svg
          aria-hidden="true"
          class="events-none icon inline-block align-t"
          data-testid="icon-lock"
          focusable="false"
          style="width: 1.3em;"
        >
          <use
            xlink:href="#icon-lock"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
        >
          lock
        </span>
      </div>
      <div
        class="flex-child-grow"
      >
        <span
          class="ml6"
        >
          Nope.
        </span>
      </div>
    </div>
  </div>
</body>
`;

exports[`ControlAlert basic success renders as expected 1`] = `
<body>
  <div>
    <div
      class="round px24 py24 flex bg-green-faint"
    >
      <div
        class="flex-child-no-shrink"
      >
        <svg
          aria-hidden="true"
          class="events-none icon inline-block align-t"
          data-testid="icon-check"
          focusable="false"
          style="width: 1.3em;"
        >
          <use
            xlink:href="#icon-check"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          />
        </svg>
        <span
          style="position: absolute; border: 0px; width: 1px; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;"
        >
          check
        </span>
      </div>
      <div
        class="flex-child-grow"
      >
        <span
          class="ml6"
        >
          Woohoo!
        </span>
      </div>
    </div>
  </div>
</body>
`;
